HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -2532,21 +2525,32 @@ def self.create_supply_fan(model, obj_name, fan_watts_per_cfm, fan_cfms)
 
     fan_cfms.sort.each do |fan_cfm|
       fan_ratio = fan_cfm / max_fan_cfm
-      power_fraction = calculate_fan_power_from_curve(1.0, fan_ratio)
+      power_fraction = calculate_fan_power_from_curve(1.0, fan_ratio, hvac_system)
       fan.addSpeed(fan_ratio.round(5), power_fraction.round(5))
     end
 
     return fan
   end
 
-  # TODO
+  # Calculate fan power at any speed or mode
   #
-  # @param max_fan_power [TODO] TODO
-  # @param fan_ratio [TODO] TODO
-  # @return [TODO] TODO
-  def self.calculate_fan_power_from_curve(max_fan_power, fan_ratio)
-    # Cubic relationship fan power curve
-    return max_fan_power * (fan_ratio**3)
+  # @param max_fan_power [Double] Rated fan power consumption
+  # @param fan_ratio [Double] Fan cfm ratio to full speed
+  # @param hvac_system [HPXML::HeatingSystem or HPXML::CoolingSystem or HPXML::HeatPump] The HPXML HVAC system of interest
+  # @return [Double] Fan power at any speed or mode
+  def self.calculate_fan_power_from_curve(max_fan_power, fan_ratio, hvac_system)
+    if hvac_system.fan_motor_type.nil?
+      # Cubic relationship fan power curve
+      fan_power = max_fan_power * (fan_ratio**3)
```

Comment:
What are the cases where the fan motor type is unknown and not defaulted? I guess I thought we'd always be defaulting upstream, in which case this code can go away because it's also PSC or BPM.